File: app/src/main/java/com/stedi/multitouchpaint/dialogs/WaitDialog.java
package com.stedi.multitouchpaint.dialogs;

import android.app.Fragment;
import android.app.FragmentManager;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.stedi.multitouchpaint.R;

public class WaitDialog extends BaseDialog {
    private static final String TAG = WaitDialog.class.getName();

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        setCancelable(false);
        getDialog().setCancelable(false);
        getDialog().setCanceledOnTouchOutside(false);
        return inflater.inflate(R.layout.wait_dialog, container, false);
    }

    public static WaitDialog start(FragmentManager fm) {
        WaitDialog dlg = new WaitDialog();
        dlg.show(fm, TAG);
        return dlg;
    }

    public static void stop(FragmentManager fm) {
        Fragment frg = fm.findFragmentByTag(TAG);
        if (frg != null && frg instanceof WaitDialog)
            ((WaitDialog) frg).dismiss();
    }
}

File: app/src/main/java/com/stedi/multitouchpaint/dialogs/WaitDialog.kt
package com.stedi.multitouchpaint.dialogs

import android.app.FragmentManager
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.stedi.multitouchpaint.R

class WaitDialog : BaseDialog() {
    companion object {
        private val TAG = WaitDialog::class.java.name

        fun show(fm: FragmentManager): WaitDialog {
            val dlg = WaitDialog()
            dlg.show(fm, TAG)
            return dlg
        }

        fun dismiss(fm: FragmentManager) {
            val frg = fm.findFragmentByTag(TAG)
            if (frg != null && frg is WaitDialog) {
                frg.dismiss()
            }
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        isCancelable = false
        dialog.setCancelable(false)
        dialog.setCanceledOnTouchOutside(false)
        return inflater.inflate(R.layout.wait_dialog, container, false)
    }
}